# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/spellchecker/TextServicesDocument.h
# Commit: 3ffbb88fe641
# Full Hash: 3ffbb88fe641031f435f31299e4eb2b844235bb4
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-07-13 21:42:32
# Regressor Bug: 1718924
# File Overlap Count: 2
# Description:
#   Bug 1718924 - part 14: Move `TextServicesDocument::SplitOffsetEntry` to `OffsetEntryArray` r=m_kato
#   
#   Depends on D119160
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119161
# ==============================================================================

diff -r 0bffa7979736 -r 3ffbb88fe641 editor/spellchecker/TextServicesDocument.h
--- a/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:46:40 2021 +0000
+++ b/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:51:46 2021 +0000
@@ -77,6 +77,17 @@
      */
     Result<EditorDOMRangeInTexts, nsresult> FindWordRange(
         nsAString& aAllTextInBlock, const EditorRawDOMPoint& aStartPointToScan);
+
+    /**
+     * SplitElementAt() splits an `OffsetEntry` at aIndex if aOffsetInTextNode
+     * is middle of the range in the text node.
+     *
+     * @param aIndex    Index of the entry which you want to split.
+     * @param aOffsetInTextNode
+     *                  Offset in the text node.  I.e., the offset should be
+     *                  greater than 0 and less than `mLength`.
+     */
+    nsresult SplitElementAt(size_t aIndex, uint32_t aOffsetInTextNode);
   };
 
   RefPtr<dom::Document> mDocument;
@@ -324,7 +335,6 @@
   bool SelectionIsValid() const;
 
   nsresult RemoveInvalidOffsetEntries();
-  nsresult SplitOffsetEntry(size_t aTableIndex, uint32_t aOffsetIntoEntry);
 };
 
 }  // namespace mozilla
